compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -161,14 +186,28 @@ static FailureOr<Path> findAMDAIETool(std::string toolName,
 std::pair<std::string, std::vector<std::string>> makeChessArgs(Path &vitisDir,
                                                                Path &tempDir,
                                                                bool verbose) {
+  std::string npuVersion = *getNPUVersion();
+  std::string archVersion;
+  std::string modelDir;
+  if (npuVersion == "npu1") {
+    archVersion = "20";
+    modelDir = "aie_ml";
+  } else if (npuVersion == "npu4") {
+    archVersion = "21";
+    modelDir = "aie2p";
+  } else {
+    llvm::errs() << "unsupported NPU version: " << npuVersion;
+    llvm::report_fatal_error("unsupported NPU version");
+  }
+
   Path aieToolsDir = vitisDir / "aietools";
   std::vector<std::string> flags{
       // -j <threads> : parallel compilation (function + file level)
-      "-j4",
+      "-j1",
```

Comment:
Why is this? You had issues with -j4?

Reply:
Yup - OOM

Reply:
Yeah.. at some point I saw something similar on 2023.x and then with 2024 it started working